includes/class-convertkit-broadcasts-importer.php

```diff
@@ -274,21 +274,49 @@ private function parse_broadcast_content( $broadcast_content ) {
 
 		$content = $broadcast_content;
 
-		// Remove open tracking.
-		$content = str_replace( '<img src="https://preview.convertkit-mail2.com/open" alt="">', '', $content );
+		// Wrap content in <html>, <head> and <body> tags with an UTF-8 Content-Type meta tag.
+		// Forcibly tell DOMDocument that this HTML uses the UTF-8 charset.
+		// <meta charset="utf-8"> isn't enough, as DOMDocument still interprets the HTML as ISO-8859, which breaks character encoding
+		// Use of mb_convert_encoding() with HTML-ENTITIES is deprecated in PHP 8.2, so we have to use this method.
+		// If we don't, special characters render incorrectly.
+		$content = '<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"></head><body>' . $content . '</body></html>';
+
+		// Load the HTML into a DOMDocument.
+		libxml_use_internal_errors( true );
+		$html = new DOMDocument();
+		$html->loadHTML( $content );
 
-		// Remove <style> elements.
-		$content = preg_replace( '/(<style *?>.*?<\/style>)/is', '', $content );
-		$content = preg_replace( '/(<style>.*?<\/style>)/is', '', $content );
+		// Load DOMDocument into XPath.
+		$xpath = new DOMXPath( $html );
+
+		// Remove certain elements and their contents, as we never want these to be included in the WordPress Post.
+
+		// Remove open tracking.
+		foreach ( $xpath->query( '//img[@src="https://preview.convertkit-mail2.com/open"]' ) as $node ) {
+			$node->parentNode->removeChild( $node ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		}
 
 		// Remove blank contenteditable table cells.
-		$content = str_replace( '<td contenteditable="false"></td>', '', $content );
+		foreach ( $xpath->query( '//td[@contenteditable="false"]' ) as $node ) {
+			$node->parentNode->removeChild( $node ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		}
 
-		// Extract just the permitted HTML.
-		$content = $this->get_permitted_html( $content, $this->broadcasts_settings->no_styles() );
```

Comment:
Am I read this right? Did removing styles formerly fall under a user-controlled setting? If so, shouldn't we preserve that?